src/constraint/encoding.ts

```diff
@@ -194,7 +194,22 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       }
       return true; // other channel is irrelevant to this constraint
     }
-  }, {
+  },{
+    name: 'barMarkWithScaleZero',
+    description: 'Do not reccommend bar mark if scale does not start at zero',
+    properties: [Property.MARK, Property.SCALE_ZERO, Property.CHANNEL, Property.TYPE],
+    requireAllProperties: true,
+    strict: true,
+    satisfy: (encQ: EncodingQuery, schema: Schema, stats: Stats, opt: QueryConfig) => {
+      if (contains([Channel.X, Channel.Y], encQ.channel) && (encQ.type === Type.QUANTITATIVE) &&
+          (encQ.type === Type.QUANTITATIVE) && encQ.scale && !encQ.bin) {
```

Comment:
This constraint have *many* issues.  

1) You have `mark` a part of the constraint.  Can it be encoding constraint?  For that reason, you haven't refer to mark at all.

2) why do you have to test `(encQ.type === Type.QUANTITATIVE)` twice?  

also why do you have to test for `encQ.bin` (which is probably irrelevant?)

If you really need to test bin here, you should add it to be a part of `properties`.  (Don't because it's irrrelvant -- unless you can prove that I overlook anything.)